### DISCUSSION TITLE
Auto Dark Mode doesn't automatically switch my theme as scheduled in the app

### DISCUSSION THREAD

**Discussion by stepuntik** [AutoDarkMode/Windows-Auto-Night-Mode#497] [Q&A] [answered]
I have set up morning and evening time to switch to Light and Dark themes respectively. What I've noticed is that themes never actually change automatically. Why is that? Can it be fixed? So far I've only benefitted from using hotkeys to switch between themes quickly, but it always has to be done manually... Please advise and help fix it, thanks

**Comment by Armin2208** [comment]
do you use a battery powered device?

**Reply by stepuntik** [reply]
Yes, I use this app on my laptop. The auto-switch doesn't occur in either mode (battery or plug-in). I literally just tested it again, and it does nothing.

**Reply by Armin2208** [reply]
have you activated "enable dark mode when device is unplugged" in adm switch modes?

**Reply by stepuntik** [reply]
No, because I don't want my laptop to switch to Dark mode during the day. I just want it to automatically switch between themes according to the set time. That's one of the main features of the app, but it doesn't work

**Reply by Armin2208** [reply]
Please send us your service.log and config.yaml file. You will find both in the config folder. You can reach it by the Settings in Auto Dark Mode.

**Comment by stepuntik** [comment]
[requested-files.zip](https://github.com/AutoDarkMode/Windows-Auto-Night-Mode/files/9330746/requested-files.zip)

Sure, here you go

**Comment by Armin2208** [comment]
I see you always forced the theme. So I expect your dark theme has dark mode set and your light theme has light mode set.
If you use the keyboard shortcuts to force the theme, Auto Dark Mode will not switch the theme anymore at next switch, until you disable the force mode.

**Reply by Armin2208** [reply]
Maybe you even press accidentally the keyboard shortcut? Because I see it really often in your log

**Reply by stepuntik** [reply]
I've been playing around with the hotkey mode just to test how quickly the system updates the UI and to see how the selected themes would look like, that's probably why you see so much of it in the logs. But why would manual selection disable auto switch? Inside of the app it's not reflected in any way, my auto switch schedule is in the same enabled mode. Also, how do you "disable force mode"?
Would it be possible to have the app being able to retain auto switch mode even if a user tested hotkeys prior to the auto switching time?

**Reply by stepuntik** [reply]
*also, it's morning time where I am right now, I just opened my laptop to discover that the dark theme is still on, even though the app is scheduled to change theme to light automatically at 7 am. That another reason why you see so many forced theme switches in the logs - I simply had to change them myself.

**Reply by Armin2208** [reply]
Do you shut down your PC or only put it into sleep mode?

**Comment by Spiritreader** [comment, marked as answer]
Please update to the latest beta version and try again.
Also toggle "Monitor active theme" on the theme page and check if that changes anything

Would also help to know what Windows version you're currently using.

**Reply by stepuntik** [reply]
Hi! 
I've reinstalled the app to the latest beta version (10.2.0.17)
I couldn't find the toggle on the theme page (there was none), so I changed the value to 'true' in the config file.
I'm currently using Windows 10 version 21H2 build 19044.1889

I'm yet to discover whether this has helped. I will update you as soon as I properly test it.

**Reply by stepuntik** [reply]
I just tested it - the auto-switch seems to be working now, including when the laptop is closed (I open it and it updates the theme according to schedule). I'll be testing it again just to be sure and will let you know if anything goes wrong.

Thanks to all of you for your timely responses and willingness to assist!
